balancer-js/src/modules/data/pool/subgraph.ts

```diff
@@ -1,46 +1,97 @@
+import { Findable } from '../types';
+import {
+  createSubgraphClient,
+  SubgraphClient,
+  SubgraphPool,
+  Pool_OrderBy,
+  OrderDirection,
+} from '@/modules/subgraph/subgraph';
+import { PoolAttribute } from './types';
 import { Pool, PoolType } from '@/types';
-import { PoolAttribute, PoolRepository } from './types';
-import { PoolToken, SubgraphClient } from '@/modules/subgraph/subgraph';
 
-export class SubgraphPoolRepository implements PoolRepository {
-  constructor(private client: SubgraphClient) {}
+/**
+ * Access pools using generated subgraph client.
+ *
+ * Balancer's subgraph URL: https://thegraph.com/hosted-service/subgraph/balancer-labs/balancer-v2
+ */
+export class PoolsSubgraphRepository implements Findable<Pool, PoolAttribute> {
+  private client: SubgraphClient;
+  public pools: SubgraphPool[] = [];
+
+  constructor(url: string) {
+    this.client = createSubgraphClient(url);
+  }
+
+  async fetch(): Promise<SubgraphPool[]> {
+    const { pool0, pool1000 } = await this.client.Pools({
+      where: { swapEnabled: true, totalShares_gt: '0' },
+      orderBy: Pool_OrderBy.TotalLiquidity,
+      orderDirection: OrderDirection.Desc,
+    });
+
+    // TODO: how to best convert subgraph type to sdk internal type?
```

Comment:
We should have some converter function/class as we'll need it to ensure all PoolRepositories return consistent data. I think the basic functionality should be to convert anything that can be converted without needing additional outside data and runs syncronously.

Reply:
I added `mapType` function for type mapping. It's just taking fields from subgraph client type and sets them for the internal Pool type.